Sources/NIO/Bootstrap.swift

```diff
@@ -146,54 +142,80 @@ public final class ServerBootstrap {
     /// - parameters:
     ///     - address: The `SocketAddress` to bind on.
     public func bind(to address: SocketAddress) -> EventLoopFuture<Channel> {
-        return bind0(eventLoopGroup: group, to: address)
+        return bind0({ address })
     }
 
     /// Bind the `ServerSocketChannel` to a UNIX Domain Socket.
     ///
     /// - parameters:
     ///     - unixDomainSocketPath: The _Unix domain socket_ path to bind to. `unixDomainSocketPath` must not exist, it will be created by the system.
     public func bind(unixDomainSocketPath: String) -> EventLoopFuture<Channel> {
-        let evGroup = group
+        return bind0 {
+            try SocketAddress(unixDomainSocketPath: unixDomainSocketPath)
+        }
+    }
+
+    /// Use the existing bound socket file descriptor.
+    ///
+    /// - parameters:
+    ///     - descriptor: The _Unix file descriptor_ representing the bound stream socket.
+    public func withBoundSocket(descriptor: CInt) -> EventLoopFuture<Channel> {
+        return bind0(makeServerChannel: { (eventLoop, childEventLoopGroup) in
+            try ServerSocketChannel(descriptor: descriptor, eventLoop: eventLoop, group: childEventLoopGroup)
+        }) { (eventLoop, serverChannel) in
+            let promise: EventLoopPromise<Void> = eventLoop.newPromise()
+            serverChannel.registerAlreadyConfigured0(promise: promise)
+            return promise.futureResult
+        }
+    }
+
+    private func bind0(_ makeSocketAddress: () throws -> SocketAddress) -> EventLoopFuture<Channel> {
+        let address: SocketAddress
         do {
-            let address = try SocketAddress(unixDomainSocketPath: unixDomainSocketPath)
-            return bind0(eventLoopGroup: evGroup, to: address)
-        } catch let err {
-            return evGroup.next().newFailedFuture(error: err)
+            address = try makeSocketAddress()
+        } catch {
+            return group.next().newFailedFuture(error: error)
+        }
+        return bind0(makeServerChannel: { (eventLoop, childEventLoopGroup) in
+            return try ServerSocketChannel(eventLoop: eventLoop,
+                                           group: childEventLoopGroup,
+                                           protocolFamily: address.protocolFamily)
```

Comment:
Similarly here it might be nicer to declare this closure as an inner function and pass it by name.

Reply:
Same thing, it captures the address.